File: app/api/schemas/health.py
from pydantic import BaseModel, Field


class HealthResponse(BaseModel):
    status: str = Field(examples=["ok"])
    service: str
    version: str
    environment: str


class DependencyStatus(BaseModel):
    status: str
    detail: str | None = None


class ReadinessResponse(BaseModel):
    status: str = Field(examples=["ready"])
    dependencies: dict[str, DependencyStatus]

File: types/api/health.ts
export interface HealthResponse {
  status: string;
  service: string;
  version: string;
  environment: string;
}

export interface DependencyStatus {
  status: string;
  detail: string | null;
}

export interface ReadinessResponse {
  status: string;
  dependencies: Record<string, DependencyStatus>;
}
